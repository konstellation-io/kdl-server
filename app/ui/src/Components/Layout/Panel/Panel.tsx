import { CSSTransition, TransitionGroup } from 'react-transition-group';
import React, { FC, memo } from 'react';

import { Button } from 'kwc';
import IconClose from '@material-ui/icons/Close';
import cx from 'classnames';
import styles from './Panel.module.scss';

export enum PANEL_THEME {
  DEFAULT = 'themeDefault',
  DARK = 'themeDark',
  LIGHT = 'themeLight',
}

export enum PANEL_SIZE {
  DEFAULT = 'default',
  BIG = 'big',
}

type Props = {
  title?: string;
  show: boolean;
  close: () => void;
  children?: JSX.Element | null | false;
  size?: PANEL_SIZE;
  noShrink?: boolean;
  theme?: PANEL_THEME;
};
const Panel: FC<Props> = ({
  show,
  close,
  children,
  title = '',
  size = PANEL_SIZE.DEFAULT,
  noShrink = false,
  theme = PANEL_THEME.DEFAULT,
}) => (
  <TransitionGroup
    className={cx(styles.group, styles[size], {
      [styles.noShrink]: noShrink,
    })}
  >
    <CSSTransition
      key={`${show}`}
      timeout={500}
      classNames={{
        enter: styles.enter,
        exit: styles.exit,
      }}
    >
      <>
        {show && (
          <div className={cx(styles.container, styles[size], styles[theme])}>
            {title !== '' && (
              <header>
                <div className={styles.separator} />
                <p className={styles.title}>{title}</p>
                <Button label="" Icon={IconClose} onClick={close} />
              </header>
            )}
            <div className={styles.content}>{children}</div>
          </div>
        )}
      </>
    </CSSTransition>
  </TransitionGroup>
);

export default memo(Panel);
