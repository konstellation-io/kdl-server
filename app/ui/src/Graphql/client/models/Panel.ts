import { PANEL_SIZE, PANEL_THEME } from 'Components/Layout/Panel/Panel';
import { GetRuntimes_runtimes } from '../../queries/types/GetRuntimes';

export type PanelInfo = {
  id: PANEL_ID;
  title: string;
  fixedWidth: boolean | null;
  theme: PANEL_THEME;
  size: PANEL_SIZE;
  runtime?: GetRuntimes_runtimes;
};

export enum PANEL_ID {
  SETTINGS = 'settings',
  PROJECT_DESCRIPTION = 'projectDescription',
  MEMBER_INFO = 'memberInfo',
  RUNTIMES_LIST = 'runtimesList',
  RUNTIME_INFO = 'runtimeInfo',
}
