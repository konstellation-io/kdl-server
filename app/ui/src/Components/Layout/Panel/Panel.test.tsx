import Panel, { PANEL_SIZE } from './Panel';

import { Button } from 'kwc';
import * as React from 'react';
import { shallow } from 'enzyme';

const onCloseMock = jest.fn();

const component = shallow(
  <Panel title="Some title" show={true} close={onCloseMock} size={PANEL_SIZE.DEFAULT} noShrink={true} dark={false}>
    <div>Children</div>
  </Panel>,
);

describe('Panel component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('show right texts', () => {
    // Arrange.
    // Act.
    // Assert.
    expect(component.contains('Some title')).toBeTruthy();
  });

  it('handles events', () => {
    // Arrange.
    // Act.
    component.find(Button).simulate('click');

    // Assert.
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('hides content when closed', () => {
    // Arrange.
    // Act.
    component.setProps({ show: false });

    // Assert.
    expect(component.find('.container').isEmptyRender()).toBeTruthy();
  });
});
