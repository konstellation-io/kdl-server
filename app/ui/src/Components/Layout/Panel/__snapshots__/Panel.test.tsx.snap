// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component Component match snapshot 1`] = `
<TransitionGroup
  childFactory={[Function]}
  className="group default noShrink"
  component="div"
>
  <CSSTransition
    classNames={
      Object {
        "enter": "enter",
        "exit": "exit",
      }
    }
    key="true"
    timeout={500}
  >
    <div
      className="container default themeDefault"
    >
      <header>
        <div
          className="separator"
        />
        <p
          className="title"
        >
          Some title
        </p>
        <P
          Icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label=""
          onClick={[MockFunction]}
        />
      </header>
      <div
        className="content"
      >
        <div>
          Children
        </div>
      </div>
    </div>
  </CSSTransition>
</TransitionGroup>
`;
